### PR TITLE
Adjust extra_services assertions

### PR DESCRIPTION
**Pull request by smola**
## Description

* Allows additional extra_services beyond the one added by the test. This can happen peer services like kafka, mysql, etc.
* Ensure that once the extra_service of interest is seen, it is also seen in all subsequent requests.

## Motivation

JIRA issue: [APPSEC-11886]

## Workflow

1. ⚠️⚠️ Create your PR as draft
2. Follow the style guidelines of this project (See [how to easily lint the code](https://github.com/DataDog/system-tests/blob/main/docs/edit/lint.md))
3. Work on you PR until the CI passes (if something not related to your task is failing, you can ignore it)
4. Mark it as ready for review

Once your PR is reviewed, you can merge it! :heart:

## Reviewer checklist

* [ ] Check what scenarios are modified. If needed, add the relevant label (`run-parametric-scenario`, `run-profiling-scenario`...). If this PR modifies any system-tests internal, then add the `run-all-scenarios` label ([more info](https://github.com/DataDog/system-tests/blob/main/docs/CI/labels.md)). 
* [ ] CI is green
   * [ ] If not, failing jobs are not related to this change (and you are 100% sure about this statement)
* [ ] if any of `build-some-image` label is present
  1. is the image labl have been updated ? 
  2. just before merging, locally build and push the image to hub.docker.com
* [ ] if a scenario is added (or removed), add (or remove) it in system-test-dasboard nightly

[APPSEC-11886]: https://datadoghq.atlassian.net/browse/APPSEC-11886?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by jandro996 — APPROVED** [review]

File: tests/remote_config/test_remote_configuration.py [hunk 1]
```diff
@@ -301,7 +301,8 @@ def remote_config_asm_extra_services_available(data):
 
     @bug(library="dotnet", reason="APPSEC-11886")
```
Comment (robertomonteromiguel) [inline tests/remote_config/test_remote_configuration.py:302]: Would these changes fix the problems with .net?

Comment (robertomonteromiguel) [reply tests/remote_config/test_remote_configuration.py:302]: @daniel-romano-DD

**Review by robertomonteromiguel — CHANGES_REQUESTED** [review]
Only to know if we can remove @bug for dotnet

File: tests/remote_config/test_remote_configuration.py (hunk 1 above)
Comment (gnufede) [reply tests/remote_config/test_remote_configuration.py:302]: @robertomonteromiguel I believe that is unrelated to this PR, why don't we test that in a different one? https://github.com/DataDog/system-tests/pull/1782

Comment (smola) [reply tests/remote_config/test_remote_configuration.py:302]: As far as I can tell, APPSEC-11886 describes exactly this issue, so I will update the PR.

**Review by gnufede — APPROVED** [review]

**Review by robertomonteromiguel — APPROVED** [review]

**Pull request merged**